### PR TITLE
Added an option in lrolola2isis to sort the output file so order of points will be consistent for testing

### PR DESCRIPTION
**Pull request by dcookastro**
## Description

An apptest was added for lrolola2isis.  lrolola2isis.xml, LidarData, and LidarControlPoint were also modified to allow a sort option when an output file is created.  This option allows the output LidarControlPoints to be sorted by ID and provide a consistent order of the control points for comparing test output to the truth data.

## Motivation and Context

This change was made to satisfy ISIS3 development requirements for testing applications.
## How Has This Been Tested?
Visual inspection and repeated runs of the new application test.

No noticeable change to other areas of the code.  The existing behavior was left as the default.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/LidarData/LidarData.cpp [hunk 1]
```diff
@@ -54,10 +53,20 @@ namespace Isis {
    *
    * @return @b QList<QSharedPointer<LidarControlPoint>> Returns list of Lidar control points.
```
Comment (kledmundson) [inline isis/src/base/objs/LidarData/LidarData.cpp:55]: should there be a better description here, e.g. explaining the bool argument and why the sorting stuff was added?

**Comment by dcookastro** [comment]
Yes.  I will.  I'll add the ticket number to the history comments as well.

**Comment by dcookastro** [comment]
It looks like the updates I just made to the documentation were added to
the pull request.

File: isis/src/base/objs/LidarData/LidarData.cpp [hunk 2]
```diff
@@ -50,14 +49,32 @@ namespace Isis {
 
 
   /**
-   * Gets the list of Lidar data points.
+   * Gets the list of Lidar data points optionally sorted . 
    *
+   * @sort An option to sort the list.  The default is false (no sort).
```
Comment (kledmundson) [inline isis/src/base/objs/LidarData/LidarData.cpp]: should the sort argument have @param in front of it? That's what I see in other method headers.

File: isis/src/base/objs/LidarData/LidarData.cpp [hunk 3]
```diff
@@ -344,10 +361,19 @@ namespace Isis {
    * @throws IException::User Throws User exception if it cannot open the file for writing.
    */
   void LidarData::write(FileName outputFile, LidarData::Format format) {
```
Comment (kledmundson) [inline isis/src/base/objs/LidarData/LidarData.cpp:363]: add @param for each of the arguments? Also maybe an explanation about the "Test" format to support a unit test? That initially confused me.

File: isis/src/base/objs/LidarData/LidarData.cpp [hunk 4]
```diff
@@ -344,10 +361,19 @@ namespace Isis {
...
+    bool sort = false;
```
Comment (kledmundson) [inline isis/src/base/objs/LidarData/LidarData.cpp]: Do you need this bool? I don't see it used anywhere but in "points(sort). Can you just use the "false" directly and put a comment explaining the this means the points are not being serialized in any particular order?

File: isis/src/base/objs/LidarData/LidarData.cpp [hunk 5]
```diff
@@ -344,10 +361,19 @@ namespace Isis {
...
     // Set up the output file
+
+    std::cout << "Formats are:  Json = " << Json << ", Binary = " << Binary << " and Test = " << Test << std::endl;
```
Comment (kledmundson) [inline isis/src/base/objs/LidarData/LidarData.cpp]: are the std::cout lines in this method meant to be here or are they just for debugging?

File: isis/src/lro/apps/lrolola2isis/tsts/twoImage/Makefile [hunk 6]
```diff
@@ -0,0 +1,23 @@
+APPNAME = lrolola2isis
+# This test exercises the ingestion of Lola data  the LRO mission.
+# This test is not complete.  The output still needs to be sorted by id and each point entry
```
Comment (kledmundson) [inline isis/src/lro/apps/lrolola2isis/tsts/twoImage/Makefile]: should this comment be updated now (sorting is complete, right?)

**Pull request merged**